### PR TITLE
chore: Publish client v0.24.0

### PR DESCRIPTION
**Pull request by liuverta**
## Impact and Context

Shipping model schema validation and custom prediction headers.

## Risks and Area of Effect

Low; largely new functionality, and model predictions have undergone validation to check for regressions.

## Testing
- [x] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

A couple of unit tests are failing, but it's unrelated to pending client changes [[Slack](https://vertaai.slack.com/archives/C01RN3BFUMD/p1690835393743289)].

## Reverting
- [ ] Contains Migration - _Do Not Revert_

Revert this PR and yank `verta==0.24.0` from PyPI.

### CODE REVIEW

File: client/verta/docs/changelog.rst [hunk 1]
```diff
@@ -37,6 +37,44 @@ Changelog
...
+  `#3940 <https://github.com/VertaAI/modeldb/pull/3940>`__)
+
+Internal Changes
+^^^^^^^^^^^^^^^^
+- remove header caching
+  (`#3941 <https://github.com/VertaAI/modeldb/pull/3941>`__)
+- remove ``model_packaging`` from model API
+  (`#3881 <https://github.com/VertaAI/modeldb/pull/3881>`__)
+
+
```
Comment (liuverta) [inline client/verta/docs/changelog.rst:77]: Screenshot 2023-07-31 at 12 19 57 PM

File: client/verta/verta/registry/entities/_modelversion.py [hunk 2]
```diff
@@ -1671,6 +1674,8 @@ def get_dataset_versions(self):
         """
         Gets all DatasetVersions associated with this model version.
 
+        .. versionadded:: 0.24.0
+
```
Comment (liuverta) [inline client/verta/verta/registry/entities/_modelversion.py:1678]: Screenshot 2023-07-31 at 12 21 00 PM

File: client/verta/verta/registry/entities/_modelversion.py [hunk 3]
```diff
@@ -52,6 +52,9 @@ class RegisteredModelVersion(_deployable_entity._DeployableEntity):
     """
     Object representing a version of a registered model.
 
+    .. versionadded:: 0.24.0
+        ``experiment_run_id`` attribute.
+
```
Comment (liuverta) [inline client/verta/verta/registry/entities/_modelversion.py:57]: Screenshot 2023-07-31 at 12 22 17 PM

File: client/verta/verta/registry/entities/_model.py [hunk 4]
```diff
@@ -1111,6 +1112,8 @@ def get_pii(self):
         """
         Returns the PII value of this registered model.
 
+        .. versionadded:: 0.24.0
+
```
Comment (liuverta) [inline client/verta/verta/registry/entities/_model.py:1116]: Screenshot 2023-07-31 at 12 22 40 PM

File: client/verta/verta/registry/entities/_model.py [hunk 5]
```diff
@@ -1092,6 +1092,7 @@ def set_pii(self, pii):
         """
         Updates the PII value of this registered model.
 
+        .. versionadded:: 0.24.0
```
Comment (liuverta) [inline client/verta/verta/registry/entities/_model.py:1096]: Screenshot 2023-07-31 at 12 23 16 PM

File: client/verta/verta/environment/_python.py [hunk 6]
```diff
@@ -25,6 +25,9 @@
 class Python(_environment._Environment):
     """Capture metadata about Python, installed packages, and system environment variables.
 
+    .. versionadded:: 0.24.0
+        ``python_version`` attribute.
+
```
Comment (liuverta) [inline client/verta/verta/environment/_python.py:30]: Screenshot 2023-07-31 at 12 23 24 PM

File: client/verta/verta/client.py [hunk 7]
```diff
@@ -71,6 +71,8 @@ class Client(object):
        The ``VERTA_DISABLE_CLIENT_CONFIG`` environment variable, when set to
        a non-empty value, disables discovery of client config files for use in
        protected filesystems.
+    .. versionadded:: 0.24.0
+        `organization_id` and `organization_name` parameters.
```
Comment (liuverta) [inline client/verta/verta/client.py:75]: Screenshot 2023-07-31 at 12 24 14 PM

File: client/verta/verta/registry/_verta_model_base.py [hunk 8]
```diff
@@ -105,6 +108,9 @@ def predict(self, input, headers: Dict[str, str]):
     def batch_predict(self, df, headers: Dict[str, str]):
         """Produce an output from `df`.
 
+        .. versionadded:: 0.24.0
+            `headers` parameter.
+
```
Comment (liuverta) [inline client/verta/verta/registry/_verta_model_base.py:113]: Screenshot 2023-07-31 at 12 24 39 PM

File: client/verta/verta/registry/_verta_model_base.py [hunk 9]
```diff
@@ -59,6 +59,9 @@ def __init__(self, artifacts):
     def predict(self, input, headers: Dict[str, str]):
         """Produce an output from `input`.
 
+        .. versionadded:: 0.24.0
+            `headers` parameter.
+
```
Comment (liuverta) [inline client/verta/verta/registry/_verta_model_base.py:64]: Screenshot 2023-07-31 at 12 24 44 PM

File: client/verta/verta/registry/_validate_schema.py [hunk 10]
```diff
@@ -14,6 +14,8 @@
 def validate_schema(f):
     """Decorator to validate prediction input and output against previously provided schema.
 
+    .. versionadded:: 0.24.0
+
```
Comment (liuverta) [inline client/verta/verta/registry/_validate_schema.py:18]: Screenshot 2023-07-31 at 12 25 14 PM

File: client/verta/verta/tracking/entities/_deployable_entity.py [hunk 11]
```diff
@@ -92,7 +92,11 @@ def _get_artifact(self, key):
 
     def log_schema(self, input: dict, output: Optional[dict] = None) -> None:
         """
-        Sets the input and output schemas, which are stored as model artifacts. To propagate this
+        Sets the input and output schemas, which are stored as model artifacts.
+
+        .. versionadded:: 0.24.0
```
Comment (liuverta) [inline client/verta/verta/tracking/entities/_deployable_entity.py:97]: Screenshot 2023-07-31 at 12 25 58 PM

Screenshot 2023-07-31 at 12 26 09 PM

File: client/verta/verta/tracking/entities/_deployable_entity.py [hunk 12]
```diff
@@ -157,6 +161,8 @@ def get_schema(self) -> Dict[str, dict]:
                 "output": <output schema>
             }
 
+        .. versionadded:: 0.24.0
```
Comment (liuverta) [inline client/verta/verta/tracking/entities/_deployable_entity.py:164]: Screenshot 2023-07-31 at 12 27 10 PM

Screenshot 2023-07-31 at 12 26 54 PM

**Review by ewagner-verta — APPROVED** [review]

**Pull request merged**